### PR TITLE
feat: add `region_statistics` table

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Add `region_statistics` table

```
mysql> select * from information_schema.region_statistics;
+---------------+----------+---------------+---------------+---------------+-----------+--------+-------------+
| region_id     | table_id | region_number | memtable_size | manifest_size | sst_size  | engine | region_role |
+---------------+----------+---------------+---------------+---------------+-----------+--------+-------------+
| 4440996184064 |     1034 |             0 |       3586817 |         11704 | 301876450 | mito   | Leader      |
| 4432406249472 |     1032 |             0 |     269686817 |         12508 | 279053485 | mito   | Leader      |
| 4445291151360 |     1035 |             0 |      56686817 |          9232 | 297195520 | mito   | Leader      |
| 4436701216768 |     1033 |             0 |     239386817 |          9139 | 281388451 | mito   | Leader      |
+---------------+----------+---------------+---------------+---------------+-----------+--------+-------------+
4 rows in set (0.01 sec)
```

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new error variant for handling region statistics.
	- Added functionality to retrieve region statistics from the cluster.
	- Expanded the information schema to include region statistics.
	- Implemented a comprehensive mechanism for managing and retrieving region statistics.

- **Bug Fixes**
	- Enhanced error handling for JSON serialization and deserialization.

- **Documentation**
	- Updated documentation to reflect the new region statistics features and error handling improvements.

### CODE REVIEW

**Comment by WenyXu** [comment]
@coderabbitai summary

**Review by MichaelScofield — APPROVED** [review]

File: src/common/meta/src/datanode.rs [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+
+                Ok(Self {
+                    timestamp_millis: time_util::current_time_millis(),
+                    cluster_id: header.cluster_id,
+                    // datanode id
+                    id: peer.id,
+                    // datanode address
+                    addr: peer.addr.clone(),
+                    rcus: region_stats.iter().map(|s| s.rcus).sum(),
+                    wcus: region_stats.iter().map(|s| s.wcus).sum(),
```
Comment (zhongzc) [inline src/common/meta/src/datanode.rs:158]: Why not `Stat::rcus`?

File: src/meta-client/src/client.rs [hunk 2]
```diff
@@ -327,6 +328,28 @@ impl ClusterInfo for MetaClient {
 
         Ok(nodes)
     }
+
+    async fn list_region_stats(&self) -> Result<Vec<RegionStat>> {
+        let cluster_client = self.cluster_client()?;
+        let range_prefix = DatanodeStatKey::key_prefix_with_cluster_id(self.id.0);
+        let req = RangeRequest::new().with_prefix(range_prefix);
+        let mut datanode_stats = cluster_client
+            .range(req)
```
Comment (zhongzc) [inline src/meta-client/src/client.rs:337]: Interesting... Use kv as API to fetch stats

**Review by zhongzc — APPROVED** [review]

File: src/meta-client/src/client.rs (hunk 2 above)
Comment (WenyXu) [reply src/meta-client/src/client.rs:337]: Yes, It's a in-memory key values store provided by metasrv.

File: src/common/meta/src/datanode.rs [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+
+use crate::error::Result;
+use crate::{error, ClusterId};
+
+pub(crate) const DATANODE_LEASE_PREFIX: &str = "__meta_datanode_lease";
+const INACTIVE_REGION_PREFIX: &str = "__meta_inactive_region";
+
+const DATANODE_STAT_PREFIX: &str = "__meta_datanode_stat";
+
+pub const REGION_STATISTIC_KEY: &str = "region_statistic";
```
Comment (fengjiachun) [inline src/common/meta/src/datanode.rs:36]:
```suggestion
pub const REGION_STATISTIC_KEY: &str = "__region_statistic";
```

Preserve tradition: "__xxx"

**Review by fengjiachun — APPROVED** [review]
Rest LGTM

File: src/catalog/src/system_schema/information_schema/region_statistics.rs [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+    async fn make_region_statistics(
+        &mut self,
+        request: Option<ScanRequest>,
+    ) -> Result<RecordBatch> {
+        let predicates = Predicates::from_scan_request(&request);
+        let mode = utils::running_mode(&self.catalog_manager)?.unwrap_or(Mode::Standalone);
+
+        match mode {
+            Mode::Standalone => {
+                // TODO(weny): implement it
```
Comment (killme2008) [inline src/catalog/src/system_schema/information_schema/region_statistics.rs:174]: Why not implement it? It's very useful for both standalone and distributed clusters to get the statistics info of regions.

Comment (fengjiachun) [reply src/catalog/src/system_schema/information_schema/region_statistics.rs:174]: I have implemented it in #4811

**Pull request merged**